js.go

```diff
@@ -1482,9 +1503,49 @@ func RateLimit(n uint64) SubOpt {
 }
 
 // BindStream binds a consumer to a stream explicitly based on a name.
-func BindStream(name string) SubOpt {
+func BindStream(stream string) SubOpt {
+	return subOptFn(func(opts *subOpts) error {
+		if opts.stream != "" {
+			return fmt.Errorf("nats: duplicate stream name (%s and %s)", opts.stream, stream)
+		}
+
+		opts.stream = stream
+		return nil
+	})
+}
+
+// BindOnly binds a subscription to a consumer without attempting to create.
+func BindOnly(params ...string) SubOpt {
 	return subOptFn(func(opts *subOpts) error {
-		opts.stream = name
+		if len(params) < 1 {
+			return ErrStreamNameRequired
+		}
+		var stream, consumer string
+		stream = params[0]
+		if stream == _EMPTY_ {
+			return ErrStreamNameRequired
+		}
+
+		// In case of pull subscribers, the durable name is a required parameter
+		// so only needed to use BindOnly with stream name.
+		if opts.cfg.Durable != _EMPTY_ {
+			consumer = opts.cfg.Durable
+		}
+		if len(params) >= 2 {
```

Comment:
How about a check for > 2 resulting in an too many params error?

Reply:
possibly also return that error when using a pull subscriber and providing two arguments

Reply:
You already would get an error if using PullSubscribe and setting both durable and different consumer name.  About too many args, I was not sure yet on adding another constant for a too many args error, could revisit later.